application/src/main/java/uk/nhs/hee/tis/revalidation/service/RecommendationTisStatusUpdateService.java

```diff
@@ -0,0 +1,87 @@
+
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationGmcOutcome.APPROVED;
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationGmcOutcome.REJECTED;
+import static uk.nhs.hee.tis.revalidation.entity.RecommendationStatus.COMPLETED;
+
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.nhs.hee.tis.revalidation.dto.RecommendationStatusCheckDto;
+import uk.nhs.hee.tis.revalidation.entity.DoctorsForDB;
+import uk.nhs.hee.tis.revalidation.entity.Recommendation;
+import uk.nhs.hee.tis.revalidation.entity.RecommendationGmcOutcome;
+import uk.nhs.hee.tis.revalidation.repository.DoctorsForDBRepository;
+import uk.nhs.hee.tis.revalidation.repository.RecommendationRepository;
+
+@Slf4j
+@Service
+public class RecommendationTisStatusUpdateService {
+
+  @Autowired
+  private RecommendationRepository recommendationRepository;
+  @Autowired
+  private DoctorsForDBRepository doctorsForDBRepository;
+  @Autowired
+  private RecommendationService recommendationService;
+  @Autowired
+  private SnapshotService snapshotService;
+
+  public void updateRecommendationAndTisStatus(
+      final RecommendationStatusCheckDto recommendationStatusCheckDto) {
+    RecommendationGmcOutcome outcome = recommendationStatusCheckDto.getOutcome();
+    setRecommendationStatusAndUpdateRepositories(recommendationStatusCheckDto.getRecommendationId(),
+        outcome);
+
+    String gmcReferenceNumber = recommendationStatusCheckDto.getGmcReferenceNumber();
+    Optional<DoctorsForDB> optionalDoctorsForDB = doctorsForDBRepository
+        .findById(gmcReferenceNumber);
+
+    //update the tis status in doctorsfordb to "complete"
+    if (optionalDoctorsForDB.isPresent() && (APPROVED.equals(outcome) || REJECTED
+        .equals(outcome))) {
+      final var doctorsForDB = optionalDoctorsForDB.get();
+      doctorsForDB.setDoctorStatus(
+          recommendationService.getRecommendationStatusForTrainee(gmcReferenceNumber));
+      doctorsForDBRepository.save(doctorsForDB);
+    }
+  }
+
+  //if gmc outcome is approved/rejected, update the relevant recommendation to approved/rejected,
+  //tis status to complete and also update the snapshot repository
+  private void setRecommendationStatusAndUpdateRepositories(final String recommendationId,
+      final RecommendationGmcOutcome recommendationGmcOutcome) {
+
+    final var optionalRecommendation = recommendationRepository.findById(recommendationId);
+    Recommendation recommendation = optionalRecommendation.get();
+
+    if (APPROVED.equals(recommendationGmcOutcome) || REJECTED.equals(recommendationGmcOutcome)) {
+      recommendation.setOutcome(recommendationGmcOutcome);
+      recommendation.setRecommendationStatus(COMPLETED);
+      recommendationRepository.save(recommendation);
+      snapshotService.saveRecommendationToSnapshot(recommendation);
```

Comment:
Will need to check if the snapshot of the recommendation already exists in the snapshot document.

Background:
The old Reval does not have the status `completed`. So all the completed recommendations migrated from the old Reval would have the `submitted_to_gmc` status to go through this cron job, but at the same time, those recommendations have got the snapshot in the DB already.
With this reason, checking of snapshot existence before saving is needed. 
![image](https://user-images.githubusercontent.com/56677534/145440713-bc379a6c-db2b-499a-94bb-d4765e10dab8.png)